### ISSUE TITLE
Allow for server-side rendering

### ISSUE THREAD

**Issue by mmcc** [muxinc/upchunk#19]
In the constructor we set up event listeners for online/offline and we do absolutely no guarding, which makes it a pain to use this in any SSR context.

**Comment by dylanjha** [comment]
Solution for using with https://nextjs.org/:

1. create a component that can imported dynamically

```jsx
// components/UploadForm.js
import * as UpChunk from '@mux/upchunk'

export default ({ }) => {
  // do your upchunk stuff in here in this component
}
```

2. Use  your page component to import the UploadForm component dynamically and not on the server with [next/dynamic](https://nextjs.org/docs/advanced-features/dynamic-import)

```jsx
// pages/upload.js
import dynamic from 'next/dynamic';

const UploadForm = dynamic(
  () => import('../components/UploadForm'),
  { ssr: false },
);

export default function UploadPage() {
  return (
      <div>
        <h1>Upload</h1>
        <UploadForm />
      </div>
  )
}
```

**Comment by dylanjha** [comment]
This patch was released in `v1.0.8`
